### ISSUE TITLE
Error when using repository methods outside of a request context

### ISSUE THREAD

**Issue by DavideCarvalho** [odavid/typeorm-transactional-cls-hooked#73]
On my open source project, we have a in memory queue, where we have some functions regards gamification stuff. So, those functions run asynchronously

Before version 0.1.17, everything worked fine, but now we are constantly getting the error below

![image](https://user-images.githubusercontent.com/15630886/101654746-9e599400-3a1f-11eb-8d6b-c9bdb499ead1.png)

As far as I can tell, it seems that when not bound to a http request context, the manager is null, and then everything stops working.

![image](https://user-images.githubusercontent.com/15630886/101654960-d82a9a80-3a1f-11eb-83c1-4969846f4c3f.png)

**Comment by odavid** [comment]
Hi @DavideCarvalho,

Can you follow: https://github.com/odavid/typeorm-transactional-cls-hooked#logging--debug and post the log?

Also, when do you patch the Repository and when do you initialize the context?

It is weird, since the `Repository.manager get/set` are doing the same thing they did before

https://github.com/odavid/typeorm-transactional-cls-hooked/blob/master/src/patch-typeorm-repository.ts#L9

**Comment by odavid** [comment]
Hi @DavideCarvalho - I could not reproduce the issue, however, I think I found a case within the code that can explain such behavior. After #61 was merged, I noticed that the `manager` can be null after a transaction was ended, but the context is still active. This seems to be a very rare case, but I restored the original behavior, so I hope it solves the issue

I would be glad if you let me know after upgrading to v0.1.18, if it still happens.
